test/0142/time-field-type.test.js

```diff
@@ -0,0 +1,363 @@
+  const oasDoc = {
+    openapi: '3.0.3',
+    components: {
+      schemas: {
+        Resource: {
+          type: 'object',
+          properties: {
+            ttl_seconds: {
+              type: 'integer',
+            },
+            offset_seconds: {
+              type: 'number',
+            },
+          },
+        },
+      },
+    },
+  };
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(0);
+  });
+});
+
+test('aep-142-time-field-type should validate _millis suffix', () => {
+  const oasDoc = {
+    openapi: '3.0.3',
+    components: {
+      schemas: {
+        Metric: {
+          type: 'object',
+          properties: {
+            timeout_millis: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+  };
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(1);
+    expect(results).toContainMatch({
+      message: 'Field "timeout_millis" should have type "integer" or "number" for duration values.',
+    });
+  });
+});
+
+test('aep-142-time-field-type should validate _micros suffix', () => {
+  const oasDoc = {
+    openapi: '3.0.3',
+    components: {
+      schemas: {
+        Timing: {
+          type: 'object',
+          properties: {
+            delay_micros: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+  };
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(1);
+    expect(results).toContainMatch({
+      message: 'Field "delay_micros" should have type "integer" or "number" for duration values.',
+    });
+  });
+});
+
+test('aep-142-time-field-type should validate _nanos suffix', () => {
+  const oasDoc = {
+    openapi: '3.0.3',
+    components: {
+      schemas: {
+        Timing: {
+          type: 'object',
+          properties: {
+            precision_nanos: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+  };
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(1);
+    expect(results).toContainMatch({
+      message: 'Field "precision_nanos" should have type "integer" or "number" for duration values.',
+    });
+  });
+});
+
+// Tests for correct timestamp fields
+
+test('aep-142-time-field-type should find no warnings for correct timestamp formats', () => {
+  const oasDoc = {
+    openapi: '3.0.3',
+    paths: {
+      '/test': {
+        get: {
+          responses: {
+            200: {
+              content: {
+                'application/json': {
+                  schema: {
+                    type: 'object',
+                    properties: {
+                      create_time: {
+                        type: 'string',
+                        format: 'date-time',
+                      },
+                      update_time: {
+                        type: 'string',
+                        format: 'date-time',
+                      },
```

Comment:
I think the following should pass but generates an error:
```suggestion
                      },
                      more_times: {
                        type: 'array',
                        items: {
                          type: 'string',
                          format: 'date-time',
                        },
                      },
```